schema/RSINTEGRATEDWATERHEATER.schema.yaml

```diff
@@ -78,123 +80,106 @@ HeatSourceType:
       Data Type: Boolean
-      Required: False
-    backup_heat_source_label: 
-      Description: "Identifier of backup heat source"
-      Data Type: "String"
-      Required: False
-    followed_by_heat_source_label: 
-      Description: "Identifier of heat source that follows"
-      Data Type: "String"
-      Required: False
-    companion_heat_source_label: 
-      Description: "Identifier of companion heat source"
-      Data Type: "String"
-      Required: False
-                                
-HeatSourceCoilConfiguration:
-  Object Type: "Enumeration"
-  Enumerators:
-    SUBMERGED:
-      Description: "Heat sources with submerged coil configuration"
-      Display Text: "Submerged"
-      Notes: ""
-    WRAPPED:
-      Description: "Heat sources with wrapped coil configuration"
-      Display Text: "Wrapped"
-      Notes: ""
-    EXTERNAL:
-      Description: "Heat sources with external coil configuration"
-      Display Text: "External"
-      Notes: ""
-       
+    backup_heat_source_id:
+      Description: "Reference to the `HeatSourceConfiguration` that engages if this one cannot meet the setpoint"
+      Data Type: "String" # TODO: Make an actual reference type (need to use 232 compliant Lattice)
+    followed_by_heat_source_id:
+      Description: "Reference to the `HeatSourceConfiguration` that should always engage after this one"
+      Data Type: "String" # TODO: Make an actual reference type (need to use 232 compliant Lattice)
+    companion_heat_source_id:
+      Description: "Reference to the `HeatSourceConfiguration` that should always engage concurrently with this one"
+      Data Type: "String" # TODO: Make an actual reference type (need to use 232 compliant Lattice)
+
+WeightedDistribution:
+  Object Type: "Data Group"
+  Data Elements:
+    normalized_height:
+      Description: "Normalized heights within the tank where the weight is defined between the bottom (0.0), and the top (1.0)"
+      Data Type: "[Numeric]"
+      Units: "-"
+      Constraints:
+        - ">= 0.0"
+        - "<= 1.0"
+      Required: True
+    weight:
+      Description: "Fraction of weight at the corresponding normalized height"
+      Data Type: "[Numeric]"
+      Units: "-"
+      Constraints:
+        - ">= 0.0"
+        - "<= 1.0"
+      Required: True
+
 HeatingLogicType:
   Object Type: "Enumeration"
   Enumerators:
-    SOC_BASED:
+    STATE_OF_CHARGE_BASED:
       Description: "State-of-charge based"
-      Display Text: "SoC based"
-      Notes: ""
-    TEMP_BASED:
+    TEMPERATURE_BASED:
       Description: "Temperature based"
-      Display Text: "Temp based"
+
+TankNodeSpecification:
+  Object Type: "Enumeration"
+  Enumerators:
+    TOP_NODE:
+      Description: "Refers to top tank node."
+      Display Text: "Top tank node"
       Notes: ""
-                          
+    BOTTOM_NODE:
+      Description: "Refers to bottom tank node."
+      Display Text: "Bottom tank node"
+      Notes: ""
```

Comment:
Where is this used? Is it simulation specific, or something the manufacturer should be able to determine?

Reply:
Make "StandbyTemperatureLocation" with "TOP_OF_TANK" and "BOTTOM_OF_TANK" options. (For now)

Reply:
Ask Ben about this.